File: testers/combined/java/generated-and-checked-in/avro_events/combined/avro_events/consumer/CustomerOrderHandler.java
package combined.avro_events.consumer;

import com.example.events.CustomerOrder;
import combined.avro_events.header.StandardHeaders;
import java.lang.Void;
import java.util.concurrent.CompletableFuture;

/** Handler interface for customer-order topic events */
public interface CustomerOrderHandler {
  /** Handle a message from the topic */
  CompletableFuture<Void> handle(
    String key,
    CustomerOrder value,
    StandardHeaders headers
  );
}

File: testers/combined/kotlin/generated-and-checked-in/avro_events/combined/avro_events/consumer/CustomerOrderHandler.kt
package combined.avro_events.consumer

import com.example.events.CustomerOrder
import combined.avro_events.header.StandardHeaders
import io.smallrye.mutiny.Uni

/** Handler interface for customer-order topic events */
interface CustomerOrderHandler {
  /** Handle a message from the topic */
  abstract fun handle(
    key: kotlin.String,
    value: CustomerOrder,
    headers: StandardHeaders
  ): Uni<Unit>
}